pkg/cmd/repo/clone/clone.go

```diff
@@ -157,7 +169,16 @@ func cloneRun(opts *CloneOptions) error {
 
 	gitClient := opts.GitClient
 	ctx := context.Background()
-	cloneDir, err := gitClient.Clone(ctx, canonicalCloneURL, opts.GitArgs)
+
+	var cloneDir string
+
+	for i := 0; i < retryLimit; i++ {
+		cloneDir, err = gitClient.Clone(ctx, canonicalCloneURL, opts.GitArgs)
```

Comment:
Although this is not trivially testable (more on that later), this looks fine in theory, but will not fix the original issue.

The original problem you've had was with `gh repo fork --clone <REPO>`, where the git clone is initiated from the `repo fork` command and is implemented here https://github.com/cli/cli/blob/79a1be4fe05f95a0f2b193f531ea91313668707f/pkg/cmd/repo/fork/fork.go#L319

The change you have applied is to the `repo clone` command, which will not affect the `repo fork` command. Furthermore, I do not think that `repo clone` command should have a retry mechanism. Therefore, I think that you should move the solution there.

Reply:
Sigh. I almost tripped on that when I did the fork default-branch-only PR, but I caught myself.